### DISCUSSION TITLE
Inpainting

### DISCUSSION THREAD

**Discussion by averad** [azuritecoin/OnnxDiffusersUI#33] [Ideas]
Inpainting UI Examples: https://github.com/averad/OnnxInpaintingUI

@azuritecoin are you open to the idea of inpainting being included as tab (t2)?

**Comment by azuritecoin** [comment]
Yeah I've been meaning to get around to adding inpainting. I got as far as adding the gradio components but I was having issues with the ImageMask component being too small. It looks like you got around that by explicitly creating your own HTML and css.

**Reply by averad** [reply]
No worries. My code is based on https://huggingface.co/spaces/multimodalart/stable-diffusion-inpainting

Could stylize the inpainting tab using html/css and update the main code in a later update.

**Reply by azuritecoin** [reply]
I can start a branch and port over some of your code.

**Comment by azuritecoin** [comment]
This is the current problem I'm having with css. Because I'm forcing a min-height, for wide images part of the image is outside of the element.

![out_of_bounds](https://user-images.githubusercontent.com/115603437/203429393-ddbe7427-5786-4609-a67f-fe72e70190bc.png)

**Reply by averad** [reply]
[Transform](https://developer.mozilla.org/en-US/docs/Web/CSS/transform): [scale()](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/scale) is an option

[Media Queries](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries) could be used for basing the layout on the users screen size.

Could always make tab2 its own special layout.

The Gradio [Image](https://gradio.app/docs/#image) help page has a good css layout (inspect the image box) using rem units and relative positioning.

**Comment by azuritecoin** [comment]
Managed to add some more inline css to resize the elements. See my commit here: https://github.com/azuritecoin/OnnxDiffusersUI/commit/ae856baebb6303fae83518d1632c615af629224f

![in_bounds](https://user-images.githubusercontent.com/115603437/203449656-17fc6d58-fbf3-4c85-871f-3c3093e75512.png)

**Reply by azuritecoin** [reply]
Nothing in that tab is connected to anything. I only placed the gradio components on the page. If you press generate, it won't do anything useful.

**Comment by azuritecoin** [comment]
Hey, so I added in all the code needed to inpainting and connected it to the UI. Can you give it a try. Here's the commit: https://github.com/azuritecoin/OnnxDiffusersUI/commit/a9214e3299dffafea4bcc7ae31e175ff24438273

**Reply by averad** [reply]
Can do, I'll give it a try as soon as I am back at my workstation.

**Reply by averad** [reply]
Created #35 for my suggested changes

**Reply by averad** [reply]
What do you think about a checkbox to select between inpainting and inpainting_legacy pipelines.

![test](https://user-images.githubusercontent.com/640619/203512691-a4b1aba4-6091-4c49-b27a-b56f9f754cd3.png)

**Reply by azuritecoin** [reply]
Is there a big difference between legacy and current inpainting? I didn't think it's worth it to add both inpainting pipeline.

**Reply by averad** [reply]
The inpainting pipeline requires an inpanting trained model to use it (like SD-1-5-inpainting)
The legacy inpainting pipeline is used to load non inpainting trained models (like SD-1.4 or Anything v3) for inpainting

**Reply by azuritecoin** [reply]
I see. I'll add in the checkbox then.
